test/Microsoft.AspNetCore.Routing.Tests/DefaultEndpointFinderTest.cs

```diff
@@ -0,0 +1,224 @@
+﻿using Microsoft.Extensions.Logging;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Routing
+{
+    public class DefaultEndpointFinderTest
+    {
+        [Fact]
+        public void FindEndpoints_IgnoresCase_ForRouteNameLookup()
+        {
+            // Arrange
+            var endpoint1 = CreateEndpoint(new Address("home"));
+            var endpoint2 = CreateEndpoint(new Address("admin"));
+            var endpointFinder = CreateDefaultEndpointFinder(endpoint1, endpoint2);
+
+            // Act
+            var result = endpointFinder.FindEndpoints(new Address("Admin"));
+
+            // Assert
+            var endpoint = Assert.Single(result);
+            Assert.Same(endpoint2, endpoint);
+        }
+
+        [Fact]
+        public void FindEndpoints_FindsEndpoint_WithMethodInfo()
+        {
+            // Arrange
+            var lookupMethodInfo = typeof(AdminController).GetMethod("Contact");
+            var endpoint1 = CreateEndpoint(new Address(typeof(HomeController).GetMethod("Index")));
+            var endpoint2 = CreateEndpoint(new Address(typeof(HomeController).GetMethod("Contact")));
+            var endpoint3 = CreateEndpoint(new Address(lookupMethodInfo));
+            var endpointFinder = CreateDefaultEndpointFinder(endpoint1, endpoint2, endpoint3);
+
+            // Act
+            var result = endpointFinder.FindEndpoints(new Address(lookupMethodInfo));
+
+            // Assert
+            var endpoint = Assert.Single(result);
+            Assert.Same(endpoint3, endpoint);
+        }
+
+        [Fact]
+        public void FindEndpoints_FindsFirstEndpoint_WithMatchingMethodInfo()
+        {
+            // Arrange
+            var lookupMethodInfo = typeof(AdminController).GetMethod("Contact");
+            var endpoint1 = CreateEndpoint(new Address(typeof(HomeController).GetMethod("Index")));
+            var endpoint2 = CreateEndpoint(new Address(lookupMethodInfo));
+            var endpoint3 = CreateEndpoint(new Address(lookupMethodInfo));
+            var endpointFinder = CreateDefaultEndpointFinder(endpoint1, endpoint2, endpoint3);
+
+            // Act
+            var result = endpointFinder.FindEndpoints(new Address(lookupMethodInfo));
+
+            // Assert
+            var endpoint = Assert.Single(result);
+            Assert.Same(endpoint2, endpoint);
+        }
+
+        [Fact]
+        public void FindEndpoints_FindsMultipleEndpoints_WithMatchingName()
+        {
+            // Arrange
+            var name = "common-tag-for-all-my-section's-routes";
```

Comment:
One could write a custom finder which might do a wildcard search on the address names "admin*"